src/gtfs_rt_utils.rs

```diff
@@ -135,35 +163,7 @@ pub fn get_model_update(
     for entity in &gtfs_rt.entity {
         let entity_id = &entity.id;
         if let Some(tu) = &entity.trip_update {
-            let trip_id = tu.trip.trip_id();
-            let date = skip_fail!(tu
-                .trip
-                .start_date
-                .as_ref()
-                .ok_or_else(|| format_err!(
-                    "The date is a mandatory field to apply a trip update, cannot apply entity {}",
-                    &entity_id
-                ))
-                .and_then(
-                    |date_str| Ok(
-                        chrono::NaiveDate::parse_from_str(&date_str, "%Y%m%d").context(format!(
-                            "impossible to read date from entity {}",
-                            &entity_id
-                        ))?
-                    )
-                ));
-
-            let vj_idx =
-                skip_fail!(model
-                    .vehicle_journeys
-                    .get_idx(trip_id)
-                    .ok_or_else(|| format_err!(
-                        "impossible to find trip {} for entity {}",
-                        &trip_id,
-                        &entity_id
-                    )));
-
-            let dated_vj = DatedVehicleJourney { vj_idx, date };
+            let dated_vj = get_dated_vj(&model.vehicle_journeys, &tu.trip, entity_id)?;
```

Comment:
I think it's better to `skip_fail!` here.

The difference is that with `?` we'll stop if a `VJ` is not good, whereas I think we should only skip it (especially since the realtime data can be a bit different from the scheduled one)